src/ClientData/DataManager.cpp

```diff
@@ -105,6 +106,23 @@ void DataManager::DeselectTracepoint(const TracepointInfo& info) {
   selected_tracepoints_.erase(info);
 }
 
+void DataManager::SelectCallstackEvents(
+    const std::vector<orbit_client_protos::CallstackEvent>& selected_callstack_events) {
+  CHECK(std::this_thread::get_id() == main_thread_id_);
+  selected_callstack_events_by_thread_id_.clear();
+
+  for (auto& event : selected_callstack_events) {
+    selected_callstack_events_by_thread_id_[event.thread_id()].emplace_back(event);
+    selected_callstack_events_by_thread_id_[orbit_base::kAllProcessThreadsTid].emplace_back(event);
+  }
+}
+
+const std::vector<orbit_client_protos::CallstackEvent>& DataManager::GetSelectedCallstackEvents(
+    uint32_t thread_id) {
+  CHECK(std::this_thread::get_id() == main_thread_id_);
+  return selected_callstack_events_by_thread_id_[thread_id];
```

Comment:
Not sure if that's the intended behaviour. If thread_id does not exist in `selected_callstack_events_by_thread_id_` it will create the entry and return a const-ref to the empty vector. If that's intended maybe we could find a name that reflects that? Something like GetOrCreateSelectedCallstackEvents? If it's not intended you could consider making the function `const`.

Reply:
Ah, I just saw that you merely copied the code from somewhere else. So feel free to ignore me on this.